### ISSUE TITLE
[Design] Profile: "I live on a United States military base outside of the United States." doesn't get checked and the form doesn't change if someone clicks "Use my mailing address for my home address." in the home address form

### ISSUE THREAD

**Issue by MarciMcGuireGCIO** [department-of-veterans-affairs/va.gov-team#37490]
## Background

Updating original details with my comment below:

If I check "I live on a United States military base outside of the United States." for mailing address and save my address....

![image.png](https://images.zenhubusercontent.com/5a4e5e33a016c5b034bc0750/019f6be1-1dca-40fd-a5a1-057461a13708)

When I go to home address and click "Use my mailing address for my home address", the address changes below, but it does not check the "I live on a United States military base outside of the United States." checkbox, nor does it change to the military address form.

I actually think this is largely a frontend ticket, but some mockups for desired functionality could probably be helpful. Let me know if you have other questions.

![image.png](https://images.zenhubusercontent.com/5a4e5e33a016c5b034bc0750/8aa1e10a-5f63-43a8-b15d-4e6c68f936f7)

## Tasks
- [ ] Determine if we need any design changes here
- [ ] If not, create ticket and outline fix to be implemented by frontend
- [ ] Determine if we need any documentation changes for contact info

## Acceptance Criteria
- [ ] We have determined whether we need any design updates or not for this ticket
- [ ] Next steps for FE are outlined and ticketed
- [ ] If needed, a ticket for documentation updates has been created

**Comment by andaleliz** [comment]
@SKasimow thanks for bringing this to my attention. I've got some questions about what we know so far, which are probably more for @Samara-Strauss since Marci created this ticket:
- What are the consequences to the Veteran if they put in a military base address and don't have that box checked?
- Do we have any kind of documentation about military base addresses to use for testing this in staging? I could google them but just want to be sure something doesn't exist before I go down that road.
- We would need to have some kind of API that knows whether or not an address is a military base or not to do any kind of validation and display something on the FE to the user. Any idea of this is already part of the address validation API we use?

If there are any docs about this already, feel free to point me there (though hopefully I would've found them in my recent documentation work!) and I can figure this stuff out.

I'll be able to put an estimate on this once I know the answers to those questions above. 

Thanks!

**Comment by Samara-Strauss** [comment]
OK, I'm sorry about this -- the original wording of this ticket is super confusing. This is the issue:

If I check I live on a United States military base outside of the United States. for mailing address and save my address....

![image.png](https://images.zenhubusercontent.com/5a4e5e33a016c5b034bc0750/019f6be1-1dca-40fd-a5a1-057461a13708)

When I go to home address and click "Use my mailing address for my home address", the address changes below, but it does not check the "I live on a United States military base outside of the United States." checkbox, nor does it change to the military address form.

I actually think this is largely a frontend ticket, but some mockups for desired functionality could probably be helpful. Let me know if you have other questions.

![image.png](https://images.zenhubusercontent.com/5a4e5e33a016c5b034bc0750/8aa1e10a-5f63-43a8-b15d-4e6c68f936f7)

**Comment by andaleliz** [comment]
Here's the [mock-up](https://www.sketch.com/s/59857eb5-d9f9-4145-99d3-d9a1de2d0655/a/R1vgV1y). A desktop one exists too but mobile should be enough for FE to work from.

I would classify this as a bug. I noticed another bug. Here are steps to reproduce:

1. Edit an address and check the "I live on a military base..." box
2. Save the address
3. Click edit again
4. The box is no longer checked and the fields revert to the non-military base field set (and do not include APO/FPO/DPO).  Screenshot below.

This is ready for FE @SKasimow :) 

<img width="508" alt="Screen Shot 2022-08-30 at 1 54 08 PM" src="https://user-images.githubusercontent.com/55992116/187510613-93acf327-b6f1-440f-ae98-441ffdee892f.png">

**Comment by Samara-Strauss** [comment]
> I noticed another bug. Here are steps to reproduce:
> 
> 1. Edit an address and check the "I live on a military base..." box
> 2. Save the address
> 3. Click edit again
> 4. The box is no longer checked and the fields revert to the non-military base field set (and do not include APO/FPO/DPO). Screenshot below.

Thanks Liz! I actually found this bug as well and already reported it to Sharon. @SKasimow did you already create a ticket for the bug I outlined in Slack/Liz outlined above?

---

As for this ticket, I just want to be clear for Sharon on what the FE ticket should include:

- Liz's mockup above
- When someone goes to home address and clicks "Use my mailing address for my home address", if the mailing address is a military address, then the home address form should:
  - Automatically check the "I live on a United States military base outside of the United States." checkbox.
  - Change the form to show the fields we use for military address.
